docs/modelserving/v1beta1/llm/huggingface/multi-node/README.md

```diff
@@ -0,0 +1,315 @@
+
+~~~
+..
+      livenessProbe:
+        failureThreshold: 2
+        periodSeconds: 10
+        successThreshold: 1
+        timeoutSeconds: 5
+        initialDelaySeconds: 10
+..
+~~~
+
+## WorkerSpec and ServingRuntime
+
+To enable multi-node/multi-GPU inference,  `workerSpec` must be configured in both ServingRuntime and InferenceService. The `huggingface-server-multinode` `ServingRuntime` already includes this field and is built on **vLLM**, which supports multi-node/multi-GPU feature. Note that this setup is **not compatible with Triton**.
+Even if the `ServingRuntime` is properly configured with `workerSpec`, multi-node/multi-GPU will not be enabled unless the InferenceService also configures the workerSpec.
+
+```
+...
+  predictor:
+    model:
+      runtime: kserve-huggingfaceserver-multinode
+      modelFormat:
+        name: huggingface
+      storageUri: pvc://llama-3-8b-pvc/hf/8b_instruction_tuned  
+    workerSpec: {} # Specifying workerSpec indicates that multi-node functionality will be used     
+```
+
+## Key Configurations
+
+When using the `huggingface-server-multinode` `ServingRuntime`, there are two critical configurations you need to understand:
+
+1. **`workerSpec.tensorParallelSize`**:    
+   This setting controls how many GPUs are used per node. The GPU type count in both the head and worker node deployment resources will be updated automatically.
+
+
+2. **`workerSpec.pipelineParallelSize`**
+  This setting determines how many nodes are involved in the deployment. This variable represents the total number of nodes, including both the head and worker nodes.
+
+
+### Example InferenceService
+
+Here’s an example of an `InferenceService` configuration for a Hugging Face model:
+
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: huggingface-llama3
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: huggingface
+      storageUri: pvc://llama-3-8b-pvc/hf/8b_instruction_tuned
+    workerSpec: {
+      pipelineParallelSize: 2
+      tensorParallelSize: 1
+    }  
+```
+
+## Serve the Hugging Face LLM Model Using 2 Nodes
+
+Follow these steps to serve the Hugging Face LLM model using a multi-node setup.
+
+### 1. Create a Persistent Volume Claim (PVC)
+
+First, create a PVC for model storage. Be sure to update `%fileStorageClassName%` with your actual storage class.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: llama-3-8b-pvc
+spec:
+  accessModes:
+    - ReadWriteMany
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 50Gi
+  storageClassName: %fileStorageClassName%
+EOF
+```
+
+### 2. Download the Model to the PVC
+  
+To download the model, export your Hugging Face token (`HF_TEST_TOKEN`) as an environment variable. Replace `%token%` with your actual token.
```

Comment:
is the %string% the pattern for place holders?

I've seen on other places {{string}}

Reply:
correct, it is for placeholders.
The pattern is what I usually use but not something special. The most important thing is that the user understands what the string means. I think it is understandable for users